client/src/components/ClickableBox.tsx

```diff
@@ -28,23 +33,44 @@ const StyledImage = styled.Image`
 
 export const ClickableBox = ({
                                  name, id, imageUrl, onPress = () => {
-    }, hasIcon = true
+    }, hasIcon = true, editMode = false, onDeletePress, onEditPress
```

Comment:
I think you should add `onDeletePress` & `onEditPress` default empty lambda function, just as in `onPress`.
In this case you won't need to validate they are defined before you access them in `handleDeletePress` & `handleEditPress`.

Reply:
done